
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>ChromaClique Manual</title>
  </head>
  <body topmargin="100" leftmargin="500" rightmargin="500">
    <div title="page-header">
      <h1>ChromaClique Manual </h1>
    </div>
    <div title="page-content">
	<ul>
	    <li><h2>Overview</h2></li>
		<div> ChromaClique is a tool for the reconstruction of nucleosome profiles of individual or  subpopulations of cells. Data from NOMe-sequencing assay is used by ChromaClique for deconvolution of different nucleosome profiles (chromatypes) from cell subpopulations of one NOME-seq measurement. ChromaClique uses a maximal clique enumeration algorithm on a newly defined NOMe read graph that is able to group reads according to their nucleosome profiles. </div>
	    <li><h2>Intallation</h2></li>
		<div title="Installation">
			<h4>Dependencies</h4> ChromaClique depends on boost, gnu parallel, and cmake and samtools. You can install them with a package manager of your choice.
			<h5>Ubuntu:</h5>
			<xmp>apt-get install libncurses5-dev cmake libboost-all-dev git build-essential zlib1g-dev parallel</xmp>
			<h5>OSX 10.8.x:</h5> Please XCode and its command line tools, and with macports:
			<xmp>port install cmake boost parallel</xmp>
			<h5>Windows:</h5> ChromaClique has not been tested on Windows. The scripts depend on the bash shell, awk, and sed.

			<h4>Installation routine</h4> If you want to install ChromaClique to a non-standard directory, change it with cmake -DCMAKE_INSTALL_PREFIX=<prefix-path> ..
			<xmp>
git clone https://github.com/shounak1990/ChromaClique
cd ChromaClique
git submodule init && git submodule update
mkdir build
cd build
cmake ..
make
			</xmp>
After running the above commands, the following five executable files should be created in the build/src folder. <xmp>bamCleaner, bamChecker, GCSwitcheFinder, chromaclique, coverageCutter</xmp>
The first three are needed to preprocess the initial data.
The created executables to the <xmp>./local/bin</xmp> folder.

ChromaClique also comes with two bash scripts.
<smp> chromaCliqueFilePreprocessing.sh </smp>
<smp>chromaFinder.sh</smp>
ChromaClique and its preprocessing steps can be both run manually, or using the aforementioned bash scripts.
		</div>
	    <li><h2>Preprocessing data</h2></li>
		<div>This step is required to find the transitions from open to close nucleosome states for the distances observed in the initial reads. This is done using the following command:
		<xmp>bash chromaFilePreprocessing.sh <bamfile> <output file path> <reference sequence in fasta format></xmp>
		The file preprocessing step creates two files calles FSSwitches.txt and RS switches.txt which are needed for the next steps.
		</div>
	    <li><h2>Running ChromaClique</h2></li>
		<div>
		Chromaclique can be run either on the entire set of reads in an alignment file or on a subsection. The subsection is done using samtools. In case the subsection flag is turned on, then the beginning and the end coordinated of the subsection needs to be provided. 
		A set of files required for analysing the data can be created by turniong the "analysisFiles"  argument to 1. ChromaClique can be run on reads aligned to one only one chromosome. This is ensured by the programs "bamChecker" and "bamCleaner". moreover the chromosome id of the chromosome that the reads are aligned to need to be mentioned. This can be found using the foloowing samtools command on an indexed bamfile.
		<xmp> samtools idxstats <indexed bamfile></xmp>
		Chromaclique can be run using the following command.
		<xmp>bash chromaFinder.sh <bamfile> <reference file> <analysis files (1 or 0)> <only iteration 1 (1 or 0)> <threshold> <subsection (1 or 0)> <chromosome id of the initial reads> <start coordinate of the reads> <end coordinate of the reads></xmp>
		An example ChropmaClique run would look like:
		<xmp>bash chromaFinder.sh ./example.bam ./reference.fasta 0 0 0.05 1 chr1 70440040 70450050</xmp>
		</div>
	    <li><h2>Output</h2></li>
		<div>ChromaClique produces two output files when the "analysisFiles" argument is set to 0. One of these files is a fasta file containing the sequences of the super reads along with their start and end positions, the number of reads that were merged to creatye that particular super-read and also which strand it maps to. The second file it produces is a bam file containing all the super reads aligned to the reference.
If the "analysisFiles" argument is set to 1, then it creates two extra files called "ChromacliqueWorking.bam" and "ResultReads.txt". The first file contains all the reads that were there after the proprocessing step. The second file contains the names of the super reads along with the names of the initial reads which wer merged to create that particular super read. These files can be used to measure the accuracy of the program using simulated data. </div>
	</ul>
    </div>
  </body>
</html>
